pkg/sdk/testint/views_gen_integration_test.go

```diff
@@ -488,6 +488,31 @@ func TestInt_Views(t *testing.T) {
 		assert.NotContains(t, returnedViews, *view2)
 	})
 
+	// proves issue https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2506
+	t.Run("show view by id: same name in different schemas", func(t *testing.T) {
+		schema, schemaCleanup := createSchema(t, client, testDb(t))
+		t.Cleanup(schemaCleanup)
+
+		name := random.String()
+		id1 := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, name)
+		id2 := sdk.NewSchemaObjectIdentifier(testDb(t).Name, schema.Name, name)
+
+		request1 := sdk.NewCreateViewRequest(id1, sql)
+		request2 := sdk.NewCreateViewRequest(id2, sql)
+
+		view1 := createViewWithRequest(t, request1)
+		view2 := createViewWithRequest(t, request2)
+
+		assert.Equal(t, id1, view1.ID())
+		assert.Equal(t, id2, view2.ID())
+
+		for i := 0; i < 10; i++ {
```

Comment:
shouldn't it be better to just show that ShowByID for id1 returns view1 and id2 view2 instead of loop? With the previous InDatabase search, I guess this could pass depending on the generated IDs. Anyway, we should most likely check both views and maybe not necessarily in a loop.

Reply:
Without the loop, it failed in like 10% of runs. I remember that you said one time that this is ordered alphabetically, but I did not find anything like that in the documentation. Having the loop made it fail constantly, that's also why I started to doubt the alphabetical ordering. This loop does not cost us anything, so I would leave it, unless you have more knowledge about the predictable ordering.

Reply:
Changed to alphabetical identifiers and :prayge: